CMakeLists.txt

```
@@ -48,7 +49,21 @@ install(
   DESTINATION share/${PROJECT_NAME})
 
 ament_export_targets(${PROJECT_NAME})
-ament_export_dependencies(ament_cmake_google_benchmark benchmark osrf_testing_tools_cpp)
+ament_export_dependencies(
+  ament_cmake_google_benchmark benchmark osrf_testing_tools_cpp)
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  set(_PERFORMANCE_TEST_FIXTURE_TESTING_IN_PROJECT TRUE)
```

You can drop all of the special casing you added to `add_performance_test.cmake` and `performance_test_fixture-extras.cmake` if you do this instead:
```suggestion
  set(performance_test_fixture_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
  add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
```

---

I thought about just manually setting the `_DIR` variable, but the add_library bit was what I was missing. Awesome.